modules/server.py

```diff
@@ -21,13 +22,20 @@ def __init__(self, env, process_time=0.000001, queue_limite=100, heap_limit=4):
     def run(self):
         try:
             while True:
-                if len(self.queue.items) > 0:  # check if there is any request to be processed
+                if len(self.remaining_queue.items) > 0:
+                    request = yield self.remaining_queue.get()  # get a request from store
+                    if request.done:
+                        yield self.heap.get(request.memory) # remove trash that shouldn't be added...
+                    yield self.env.process(self.process_request(request))
+
+                elif len(self.queue.items) > 0:  # check if there is any request to be processed
                     request = yield self.queue.get()  # get a request from store
                     yield self.env.process(self.process_request(request))
 
                 yield self.env.timeout(self.process_time)  # wait for...
```

Comment:
I might be totally wrong, but does that mean all request are going ot have the same processing time?

If yes, this is not correct because even for equal requests, the processing time has a tail.

Would you mind calling this service_time (it matches to what we call in the papers, thus it might make thing easier).

Reply:
In version -1 the duration of each request was concerted to be constant. This was concerted to have as soon as possible a simple version of the simulator. The next versions will be more representative.

Reply:
By the way, the current version of the server allows different service_time, but, as Paulo said, we concerted to be constant.

Reply:
That's right. Let's move on. We just need to keep in mind that we can not cross-validate those results with the experiment results. In experiment results, even for equal requests, the process time is different (tailed).